src/displayobjects/rectangle.js

```diff
@@ -1,68 +1,239 @@
-(function(oCanvas, window, document, undefined){
+(function (oCanvas, window, document, undefined) {
 
 	// Define the class
 	var rectangle = function (settings, thecore) {
-		
+
+		function normalizeCornerBorderRadius(cornerBorderRadius, generalBorderRadius) {
+
+			if (cornerBorderRadius !== undefined) {
+				return cornerBorderRadius;
+			}
+
+			if (generalBorderRadius !== undefined) {
+				return generalBorderRadius;
+			}
+
+			return 0;
+
+		}
+
+		function normalizeBorderRadius(borderRadius, borderTopLeftRadius, borderTopRightRadius, borderBottomLeftRadius, borderBottomRightRadius) {
+
+			return {
+				topLeft: normalizeCornerBorderRadius(borderTopLeftRadius, borderRadius),
+				topRight: normalizeCornerBorderRadius(borderTopRightRadius, borderRadius),
+				bottomLeft: normalizeCornerBorderRadius(borderBottomLeftRadius, borderRadius),
+				bottomRight: normalizeCornerBorderRadius(borderBottomRightRadius, borderRadius)
+			};
+
+		}
+
+		function addAmountToBorderRadius(borderRadius, amount) {
+			if (amount > 0) {
+				return {
+					// When there is no borderRadius (=== 0), keep it as 0
+					topLeft: borderRadius.topLeft === 0 ? 0 : borderRadius.topLeft + amount,
+					topRight: borderRadius.topRight === 0 ? 0 : borderRadius.topRight + amount,
+					bottomLeft: borderRadius.bottomLeft === 0 ? 0 : borderRadius.bottomLeft + amount,
+					bottomRight: borderRadius.bottomRight === 0 ? 0 : borderRadius.bottomRight + amount
+				};
+			} else if (amount < 0) {
+				return {
+					// Never least than 0
+					topLeft: Math.max(borderRadius.topLeft + amount, 0),
```

Comment:
Could we also limit the max value to half the width/height? I'm trying out the code and I get weird rendering if the borderRadius is bigger than that:

<img width="513" alt="screen shot 2018-07-08 at 18 50 20" src="https://user-images.githubusercontent.com/382138/42422057-f1785af8-82df-11e8-99a9-e6ea869702f6.png">

As fun effects as that might be (:D) I think we should limit things to a more expected result.